src/librustdoc/html/render.rs

```diff
@@ -3458,7 +3457,7 @@ fn render_impl(w: &mut fmt::Formatter, cx: &Context, i: &Impl, link: AssocItemLi
     }
 
     let traits = &cache().traits;
-    let trait_ = i.trait_did().and_then(|did| traits.get(&did));
+    let trait_ = i.trait_did().map(|did| &traits[&did]);
```

Comment:
What's with the change here (and above)? Change the failure mode for missing DefIds?

Reply:
There now shouldn't be any cases where the trait is missing from the `traits` map so this change is to make sure it stays like that.